### PR TITLE
Add initial load tests

### PR DESCRIPTION
**Pull request by bryanhuhta**
This is the initial pass at adding load tests with k6. Right now we only are testing read APIs. To run a load test against a local Pyroscope, run:

```
./tools/k6/run.sh reads.js
```

To run a load test against a remote Pyroscope, configure your `.env` file correctly (see `tools/k6/README.md` for more details) and run:

```
./tools/k6/run.sh -c reads.js
```

To view all available load tests, run

```
./tools/k6/run.sh -h
```

### CODE REVIEW

File: tools/k6/reads.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+export default function() {
+  group('reads last 1h', function() {
+    const { start, end } = newRelativeTimeRange(1, 'h');
+    doAllQueryRequests(start, end);
+  });
+
+  group('reads last 24h', function() {
+    const { start, end } = newRelativeTimeRange(24, 'h');
+    doAllQueryRequests(start, end);
+  });
```
Comment (simonswine) [inline tools/k6/reads.js:67]: I think an environment variable for those durations would be nice

export K6_QUERY_DURATIONS=1h,24h

File: tools/k6/reads.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+function doAllQueryRequests(start, end) {
+  doSelectMergeProfileRequest({
+    start,
+    end,
+    profile_typeID: 'process_cpu:cpu:nanoseconds:cpu:nanoseconds',
+    label_selector: '{service_name="fire-dev-001/ingester"}',
```
Comment (simonswine) [inline tools/k6/reads.js:75]: I also think that would be useful to customize without code change:

```suggestion
    label_selector: '{service_name="%s"}' % (K6_QUERY_SERVICE_NAME),
```

**Review by simonswine — APPROVED** [review]
LGTM

File: tools/k6/reads.js (hunk 1 above)
Comment (bryanhuhta) [reply tools/k6/reads.js]: Added in https://github.com/grafana/pyroscope/pull/3331/commits/e00e5fd306de2847c028a665c75c367b134d8e86.

File: tools/k6/reads.js (hunk 2 above)
Comment (bryanhuhta) [reply tools/k6/reads.js]: Added in https://github.com/grafana/pyroscope/pull/3331/commits/e00e5fd306de2847c028a665c75c367b134d8e86.

**Pull request merged**